symphony/bdk/core/auth/auth_session.py

```diff
@@ -3,12 +3,32 @@ class AuthSession:
     """
 
     def __init__(self, authenticator):
-        self.session_token = None
-        self.key_manager_token = None
+        self._session_token = None
+        self._key_manager_token = None
         self._authenticator = authenticator
 
     async def refresh(self):
         """Trigger re-authentication to refresh the tokens.
         """
-        self.session_token = await self._authenticator.retrieve_session_token()
-        self.key_manager_token = await self._authenticator.retrieve_key_manager_token()
+        self._session_token = await self._authenticator.retrieve_session_token()
+        self._key_manager_token = await self._authenticator.retrieve_key_manager_token()
+
+    @property
+    async def session_token(self):
```

Comment:
and now when I see as async I'm wondering if it is the right way to go (I'm lacking experience on async in Python here :)) 
let's go this way for now and we can always reevaluate that later